src/main/java/com/hillel/items_exchange/service/impl/CategoryServiceImpl.java

```diff
@@ -1,31 +1,242 @@
 import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
 import org.springframework.stereotype.Service;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 public class CategoryServiceImpl implements CategoryService {
 
     private final CategoryRepository categoryRepository;
+    private final SubcategoryRepository subcategoryRepository;
+    private final ModelMapper modelMapper;
+    private final CategoryMapper categoryMapper;
 
     @Override
     public List<String> findAllCategoryNames() {
-        List<String> categoryNames = categoryRepository.findAllCategories();
+        List<String> categoryNames = categoryRepository.findAllCategoriesNames();
 
         if (categoryNames.isEmpty()) {
-            log.warn("IN CategoryServiceImpl (findAllCategoryNames): there are no categories");
+            log.warn("IN CategoryServiceImpl (findAllCategoryNames): there are not categories names");
             return Collections.singletonList("No categories");
         } else {
             log.info("IN CategoryServiceImpl (findAllCategoryNames): there are these categories: {}", categoryNames);
             return categoryNames;
         }
     }
+
+    @Override
+    public List<CategoryControllerDto> findAllCategories() {
+        List<CategoryControllerDto> allCategories = modelMapper.map(categoryRepository.findAll(),
+                new TypeToken<List<CategoryControllerDto>>() {
+                }.getType());
+
+        if (allCategories.isEmpty()) {
+            log.warn("IN CategoryServiceImpl (findAllCategories): there are not categories");
+            return Collections.emptyList();
+        } else {
+            log.info("IN CategoryServiceImpl (findAllCategories): there are these categories have been found: {}",
+                    allCategories);
+            return modelMapper.map(categoryRepository.findAll(), new TypeToken<List<CategoryControllerDto>>() {
+            }.getType());
+        }
+    }
+
+    @Override
+    public Optional<CategoryControllerDto> findCategoryById(Long id) {
+        if (categoryRepository.findById(id).isPresent()) {
+            log.info("IN CategoryServiceImpl (findCategoryById): the category has been found by id {}", id);
+        } else {
+            log.warn("IN CategoryServiceImpl (CategoryById): has not been found by id {}", id);
+        }
+        return categoryRepository.findById(id).map(this::mapCategoryToDto);
+    }
+
+    @Override
+    public List<String> findSubcategoryNamesByCategoryId(Long categoryId) {
+        List<String> subcategoryNames = subcategoryRepository.findSubcategoriesNamesByCategory(categoryId);
+
+        if (subcategoryNames.isEmpty()) {
+            log.warn("IN CategoryServiceImpl (findSubcategoryNamesByCategoryId): there no subcategories " +
+                    "in the category with id: {}", categoryId);
```

Comment:
Great idea to use brackets for logging, why you don't use the same on Controller layer?